### ISSUE TITLE
Prevent continuing when there is corrupt data. 

### ISSUE THREAD

**Issue by tiger5226** [lbryio/chainquery#62]
If the chainquery app panics, and it auto-restarts as a service, the block processing will continue beyond the problem. This should not happen. The app panics when there is corrupt data that needs to be fixed. 

The usual solution is to remove the corrupted blocks. We should find a way to detect and remove the corrupted blocks so a restart is successful.

**Comment by tiger5226** [comment]
So I have a branch ready for this. If block processing error'd out, the partially processed block was not rolled back. In order to roll a block back an overhaul of the concurrent transaction processing was needed so that an error could be returned and handled. The stopper pattern was implemented for all go routines at each level ( concurrent transaction, input, output processing ). Advanced debug logging was added as well to make investigations easier. Lastly, the grabbing of transaction details from LBRYcrd were serialized to allow for an algorithm that determines in which sequence transactions are assigned to available workers to be processed. Over a set of blocks, the loss of performance was minimal going from 4 min 42 sec to 4 min 44 sec. This algorithm can be improved significantly in the future.

Right now the algorithm just inspect dependencies between transactions and sorts those that have a lot of dependants to be processed by the workers first. This is a big improvement and reduces the failure rate of a transaction, which causes it to go to the back of the queue. The improvement would be to make sure, given the number of workers, that two dependant transactions don't get assigned workers at the same time. Reprocessing, eats away at the concurrency gains. So smarter assignment to workers will significantly reduce this in the future. 

The data corruption this prevents is when a block errors out during processing, it will not just go to the next block. I witnessed this in action and it corrupts the data and you have figure out where the corruption happened. This solution allows for the current block to be rolled back like a reorg. The time investment was significant but it will ensure Chainquery stays a rock solid service. So now a block, if it fails to process, will remove itself and trigger a warning, then try again. This is a looping action. Right now the max retries are set to 1000 times. So the probability of this happening twice should be very rare. I tested the solution with max retries at 0, to make sure the concurrency design works, never hits a deadlock on an error and successfully looped hundreds of times.
